src/utils.py

```diff
@@ -15,21 +200,21 @@ def csv2npz(dataset_x_path, dataset_y_path, labels_path='labels.json'):
         labels = json.load(stream_json)
 
     m = x.shape[0]
-    K = 672  # Can be found through csv
+    K = x.shape[1]  # Can be found through csv
 
     # Create R and Z
     R = x[labels["R"]].values
     R = R.astype(np.float32)
 
     X = y[[f"{var_name}_{i}" for var_name in labels["X"]
-            for i in range(K)]]
+           for i in range(K)]]
     X = X.values.reshape((m, -1, K))
     X = X.astype(np.float32)
 
     Z = x[[f"{var_name}_{i}" for var_name in labels["Z"]
-            for i in range(K)]]
+           for i in range(K)]]
     Z = Z.values.reshape((m, -1, K))
 #     Z = Z.transpose((0, 2, 1))
     Z = Z.astype(np.float32)
-    
-    np.savez('dataset.npz', R=R, X=X, Z=Z)
\ No newline at end of file
+
+    np.savez(path.join(output_path, filename), R=R, X=X, Z=Z)
```

Comment:
The path joining could be done with the pathlib instead, for instance
```python
Path(output_path).joinpath(filename)
```

Reply:
Makes sense.